brave/brave-core
# Integrate with SKU SDK (Rust) and provide safe credential interception for VPN

**NOTE: please DO review! 😄 But DO NOT merge (yet)! 
There are steps needed to be taken before this can be merged**

- [x] Thorough review by reviewers
- [x] Security review https://github.com/brave/security/issues/670
- [x] All review feedback addressed
- [x] Commits squashed where possible (reword commits to help make more useful)

Fixes https://github.com/brave/brave-browser/issues/17614
Fixes https://github.com/brave/brave-browser/issues/17254

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Visit https://account.brave.software
2. Buy VPN
3. Open Dev tools (F12 / Cmd + Opt + I)
4. Get the Order ID for the product you just purchased. You can find it in local storage tab of dev tools (the value will be a large JSON blob)
5. Go to console tab
6. Run the following in order:
    ```
    await window.chrome.braveSkus.refresh_order('YOUR_ORDER_ID_HERE')
    await window.chrome.braveSkus.fetch_order_credentials('YOUR_ORDER_ID_HERE')
    await window.chrome.braveSkus.credential_summary('vpn.brave.software')
    ```
7. Verify you get expected results (`fetch_order_credentials` will return empty string; others will return either JSON)
8. Running `credential_summary` should trigger a login to the VPN
